be/src/util/metrics.h

```diff
@@ -193,230 +174,163 @@ class CoreLocalCounter : public Metric {
     void increment(const T& delta) {
         __sync_fetch_and_add(_value.access(), delta);
     }
+
+    rj::Value to_json_value() const override {
+        return rj::Value(value());
+    }
+
 protected:
     CoreLocalValue<T> _value;
 };
 
 template<typename T>
 class AtomicCounter : public AtomicMetric<T> {
 public:
-    AtomicCounter(MetricUnit unit)
-            : AtomicMetric<T>(MetricType::COUNTER, unit) {}
-    virtual ~AtomicCounter() { }
+    AtomicCounter() {}
+    virtual ~AtomicCounter() {}
 };
 
 template<typename T>
 class AtomicGauge : public AtomicMetric<T> {
 public:
-    AtomicGauge(MetricUnit unit)
-            : AtomicMetric<T>(MetricType::GAUGE, unit) {}
-    virtual ~AtomicGauge() { }
+    AtomicGauge() : AtomicMetric<T>() {}
+    virtual ~AtomicGauge() {}
 };
 
 template<typename T>
 class LockCounter : public LockSimpleMetric<T> {
 public:
-    LockCounter(MetricUnit unit)
-      : LockSimpleMetric<T>(MetricType::COUNTER, unit) {}
-    virtual ~LockCounter() { }
+    LockCounter() : LockSimpleMetric<T>() {}
+    virtual ~LockCounter() {}
 };
 
 // This can only used for trival type
 template<typename T>
 class LockGauge : public LockSimpleMetric<T> {
 public:
-    LockGauge(MetricUnit unit)
-      : LockSimpleMetric<T>(MetricType::GAUGE, unit) {}
-    virtual ~LockGauge() { }
+    LockGauge() : LockSimpleMetric<T>() {}
+    virtual ~LockGauge() {}
 };
 
-// one key-value pair used to
-struct MetricLabel {
+using Labels = std::unordered_map<std::string, std::string>;
+struct MetricPrototype {
```

Comment:
What's the purpose of MetricPrototype?
Under which circumstance, MetricPrototype can be used to split the Metric.